sipparser/parser.go

```diff
@@ -246,11 +250,23 @@ func (s *SipMsg) addHdr(str string) {
 		case s.hdr == "X-RTP-Stat":
 			s.parseRTPStat(s.hdrv)
 		case s.hdr == "Expires":
+			s.Expires = s.hdrv
 		default:
 			if len(s.XHeader) > 0 {
 				for i := range s.XHeader {
 					if s.hdr == s.XHeader[i] {
-						s.XCallID = s.hdrv
+						var filter *regexp.Regexp
+						var ok bool
+						filter, ok = config.CompileStore.RegexMap[s.XHeader[i]]
```

Comment:
You can just use
filter, ok := config.CompileStore.RegexMap[s.XHeader[i]]

and drop the vars above

Reply:
done